File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string MentorOrMentee { get; set; }
        public int MinYearsOfCareerExperience { get; set; } = 0; // MinAge
        public int MaxYearsOfCareerExperience { get; set; } = 50; // MinAge
        public string OrderBy { get; set; } = "lastActive"; // the default sorting
    }
}

File: client/src/app/_models/userParams.ts
// a class is being used here instead of an interface to take advantage of

import { User } from "./user";

// the fact that a constructor can be used to initialize some of the properties
export class UserParams {
  mentorOrMentee: string;
  minYearsOfCareerExperience = 0;
  maxYearsOfCareerExperience = 50;
  pageNumber = 1;
  pageSize = 5;
  orderBy = 'lastActive';

  constructor(user: User) {
    this.mentorOrMentee = user.mentorOrMentee === 'Mentor' ? "Mentee" : 'Mentor';
  }

}
